doc/manual.rst

```diff
@@ -566,6 +566,19 @@ The type suffixes are:
   `'f64`             float64
 =================    =========================
 
+If the suffix is not predefined, then the suffix is assumed to be a call
+to a proc, template, macro or other callable identifier that is passed the
+string containing the literal. The identifier needs to contain the apostrophe
+and so will need to be escaped with backticks.
+
+.. code-block:: nim
+  import strutils
+  type nibble = distinct uint8
+  proc `'nibble`(n: string): uint8 =
+    result = (parseInt(n) and 0x0F).uint8
+  
+  var x = 5'nibble
```

Comment:
also add an example with extra args as you mentioned in PR:
```nim
proc `'foo(a: string, b: int) = $(a, b)
assert -12'foo(2) == ("-12", 2)
```

Reply:
Done.